### PR TITLE
Add MapKeycloakTransaction.exists methods

### LINKED ISSUE #10376: Add MapKeycloakTransaction.exists(id) method
## Description for the problem

There are multiple uses of `MapKeycloakTransaction.read(id)` across providers that only check existence of an object without actual need to read it in full. Some storages can optimize for checking the existence.

## Task

Introduce a method `exists(id)` and `exists(ModelCriteriaBuilder)` with defauilt implementations that fall back to `read(id)` and `getCount(mcb) > 0`.

### PR DESCRIPTION
**Pull request by martin-kanis**
Closes #10376

### CODE REVIEW

File: model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/userSession/HotRodUserSessionTransaction.java
Comment (mhajas) [inline model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/userSession/HotRodUserSessionTransaction.java:69]: These changes should be in the followup PR.

**Review by mhajas — CHANGES_REQUESTED** [review]
Thanks for the changes, see the comment in code

File: model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/userSession/HotRodUserSessionTransaction.java
Comment (martin-kanis) [reply model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/userSession/HotRodUserSessionTransaction.java]: done

**Review by mhajas — DISMISSED** [review]
Thanks @martin-kanis, LGTM

**Comment by mhajas** [comment]
We need to wait for the tests to run before we can merge.

**Comment by martin-kanis** [comment]
Model test failures are not cased by this PR https://github.com/keycloak/keycloak/issues/16634

**Review by mhajas — APPROVED** [review]

**Pull request merged**
